dataprofiler/profilers/profile_builder.py

```diff
@@ -1355,11 +1359,125 @@ def _update_row_statistics(self, data, sample_ids=None):
             self.row_has_null_count = len(null_in_row_count)
             self.row_is_null_count = len(null_rows)
 
+    def _get_correlation(self, clean_samples):
+        """
+        Calculate correlation matrix on the cleaned data.
+
+        :param clean_samples: the input cleaned dataset
+        :type clean_samples: dict()
+        """
+        columns = self.options.correlation.columns
+        if columns is None:
+            for col in self._profile:
+                if (self._profile[col].profile['data_type'] not in ['int', 'float']
+                        or self._profile[col].profile['statistics']['null_count'] > 0):
+                    clean_samples.pop(col)
```

Comment:
this basically means any column that has a single nan will no longer be part of corr? That's dangerous, no?

Reply:
My view about this is that we're gonna have several options: exclude columns with nan, keep those columns but with nan results, keep those columns but only calculate results based on not-nan values, keep those columns but calculate results with replaced values for nan values. Later we can think about how to deal with other cases. For now, I think we can just add some warning messages at the very least.